### PR TITLE
XRDDEV-1582

### PR DESCRIPTION
**Pull request by iluwatar**

### CODE REVIEW

**Review by olliru — APPROVED** [review]
Comments added to files. The findings do not prevent merging but may require further work,

File: src/proxy-ui-api/frontend/tests/e2e/pages/common/addSubjectsPopup.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ const commands = {
 
 const addSubjectsPopup = {
   selector:
-    '//div[contains(@class, "xrd-card") and .//span[contains(@class, "headline") and contains(text(),"Add Subjects")]]',
+    '//div[contains(@class, "xrd-card") and .//span[contains(@class, "headline") and contains(@data-test,"access-rights-dialog-title")]]',
```
[the remark this answers is not in the document]
Comment (iluwatar) [reply src/proxy-ui-api/frontend/tests/e2e/pages/common/addSubjectsPopup.js]: Good point. Addressed this in multiple places in commit https://github.com/nordic-institute/X-Road/pull/894/commits/e7f6ec37997a93def167bc2d6f84d77669f184d2

File: src/proxy-ui-api/frontend/tests/e2e/pages/serviceClients/serviceClientsPage.js [hunk 2]
```diff
@@ -72,7 +72,7 @@ module.exports = {
   sections: {
     serviceClientsTab: {
       selector:
-        '//div[contains(@class, "v-tabs-bar__content")]//a[contains(@class, "v-tab--active") and contains(text(), "service clients")]',
+        '//div[contains(@class, "v-tabs-bar__content")]//a[contains(@data-test, "serviceClients")]',
```
[the remark this answers is not in the document]
Comment (iluwatar) [reply src/proxy-ui-api/frontend/tests/e2e/pages/serviceClients/serviceClientsPage.js]: In this case the `data-test` value is generated from the Vue component's key to make it unique, so I think it's ok here. The general convention of using kebab-case doesn't fit the naming convention we use in Vue very well.

File: src/proxy-ui-api/frontend/tests/e2e/pages/tabs/keysTab.js [hunk 3]
```diff
@@ -282,23 +282,23 @@ var addKeywizardGenerateCommands = {
 const keysTab = {
   url: `${process.env.VUE_DEV_SERVER_URL}/keys`,
   selector:
-    '//div[.//a[contains(@class, "v-tab--active") and contains(text(), "Keys and certificates")]]//div[contains(@class, "base-full-width")]',
+    '//div[.//a[contains(@class, "v-tab--active") and contains(@data-test, "keys")]]//div[contains(@class, "base-full-width")]',
```
[the remark this answers is not in the document]
Comment (iluwatar) [reply src/proxy-ui-api/frontend/tests/e2e/pages/tabs/keysTab.js]: fixed

File: src/proxy-ui-api/frontend/tests/e2e/pages/ssMainPage.js [hunk 4]
```diff
@@ -936,15 +929,15 @@ module.exports = {
           locateStrategy: 'xpath',
         },
         localGroupAddMembersButton: {
-          selector: '//button[.//*[contains(text(), "Add Members")]]',
+          selector: '//button[.//*[contains(@data-test, "add-members-button")]]',
```
[the remark this answers is not in the document]
Comment (iluwatar) [reply src/proxy-ui-api/frontend/tests/e2e/pages/ssMainPage.js]: Fixed with https://github.com/nordic-institute/X-Road/pull/894/commits/90855a486b4b3834287945004b673166d53344a9

File: src/proxy-ui-api/frontend/tests/e2e/specs/ss-clients-openapiservices.js [hunk 5]
```diff
@@ -412,7 +398,7 @@ module.exports = {
 
     // Verify cancel remove all
     operationDetails.removeAllAccessRights();
-    browser.waitForElementVisible(removeAllAccessRightsPopup);
+    // browser.waitForElementVisible(removeAllAccessRightsPopup);
```
[the remark this answers is not in the document]
Comment (iluwatar) [reply src/proxy-ui-api/frontend/tests/e2e/specs/ss-clients-openapiservices.js]: Fixed with https://github.com/nordic-institute/X-Road/pull/894/commits/21ef50e2dd7cb3c601fa4de5af548372bf4f6c06

File: src/proxy-ui-api/frontend/tests/e2e/specs/ss-clients-restservices.js [hunk 6]
```diff
@@ -691,7 +642,7 @@ module.exports = {
     restEndpoints.openEndpoint('POST', '/testreq3');
     endpointPopup.deleteEndpoint();
     browser.waitForElementVisible(
-      '//*[contains(@data-test, "dialog-title") and contains(text(), "Delete endpoint")]',
+      '//div[contains(@data-test, "dialog-simple") and .//span[contains(@data-test, "dialog-title")]]',
```
[the remark this answers is not in the document]
Comment (iluwatar) [reply src/proxy-ui-api/frontend/tests/e2e/specs/ss-clients-restservices.js]: Fixed the double check with https://github.com/nordic-institute/X-Road/pull/894/commits/ced678b7274229be2d17926fe4de04f26d86c20b

**Pull request merged**